crypto/http/http_lib.c

```diff
@@ -22,8 +25,11 @@ int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
 {
     char *p, *buf;
     char *host;
-    char *port = "80";
+    const char *port = OSSL_HTTP_PORT;
+    size_t https_len = strlen(OSSL_HTTPS_NAME);
 
+    if (!ossl_assert(https_len >= strlen(OSSL_HTTP_NAME)))
```

Comment:
This seems pointless given the line just above.

Reply:
This just makes a precondition of a later pointer calculation obvious
and checks it to be on the safe side (in case the values of `OSSL_HTTP_NAME` or 'OSSL_HTTPS_NAME` changes, yeah, pretty unlikely for the next 50+ years).

Reply:
Ah...actually my comment was because I hadn't spotted that one was referring to `OSSL_HTTPS_NAME` and one was `OSSL_HTTP_NAME`...the extra `S` had passed me by. So...ok.